teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -860,55 +995,71 @@ Pros:
 Cons:
 * The line between this and a Pipeline seems very thin
 * New CRD to contend with
+* Will mean that both the Pipelines controller and this controller will need to share (or duplicate) a lot of
+  Pipeline logic
 
-### Rely on the affinity assistant
-
-In this approach we'd rely on [the affinity assistant](https://github.com/tektoncd/community/pull/318/files) to
-co-locate pods that use the same workspace.
-
-Cons:
-* Doesn't help us with the overhead of multiple pods
-* [TEP-0046](https://github.com/tektoncd/community/pull/318/files) explores shortcomings of this approach
-
-### Custom scheduler
-
-In this approach we use a custom scheduler to schedule our pods.
-
-Cons:
-* [@jlpetterson has explored this option](https://docs.google.com/document/d/1lIqFP1c3apFwCPEqO0Bq9j-XCDH5XRmq8EhYV_BPe9Y/edit#heading=h.18c0pv2k7d1a)
-  and felt it added more complexity without much gain (see also https://github.com/tektoncd/pipeline/issues/3052)
-    * For example the issue [#3049](https://github.com/tektoncd/pipeline/issues/3049)
-* Doesn't help us with the overhead of multiple pods (each Task would still be a pod)
-
-### Support other ways to share data (e.g. buckets)
+###### Custom Pipeline
 
-In this approach we could add more workspace types that support other ways of sharing data between pods, for example
-uploading to and downloading from s3. (See [#290](https://github.com/tektoncd/community/pull/290/files).)
+This approach is very similar to [Grouping CRD](#grouping-crd) but instead of creating a new type in the Tekton API,
```

Comment:
This is the exact same approach to be honest. I would honestly see the `TaskGroup` CRD to be started as a CustomTask anyway 😝